src/Orleans/Providers/ProviderTypeLoader.cs

```diff
@@ -87,6 +87,14 @@ private void ProcessAssemblyLocally(Assembly assembly)
 
         private static void ProcessNewAssembly(object sender, AssemblyLoadEventArgs args)
         {
+#if !NETSTANDARD
+            // If the assembly is loaded for reflection only avoid processing it.
+            if (args.LoadedAssembly.ReflectionOnly)
```

Comment:
Ok, sounds good. I checked and the code I was referring to happens before this is hit, so it's as expected. BTW, I was referring to: https://github.com/dotnet/orleans/blob/430a1de3c1eb1105a31133117ea4f06d2e1d01d2/src/Orleans/Runtime/OutsideRuntimeClient.cs#L224 and https://github.com/dotnet/orleans/blob/7cf2035e868e81b90767d7aff748405d01145117/src/OrleansRuntime/GrainTypeManager/SiloAssemblyLoader.cs#L73 (which first scans assemblies using reflection-only, and if it finds something that implements IProvider, it fully loads them)